pkg/bicopy/bicopy.go

```diff
@@ -26,56 +7,22 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-// Bicopy is from https://github.com/rootless-containers/rootlesskit/blob/v0.10.1/pkg/port/builtin/parent/tcp/tcp.go#L73-L104
-// (originally from libnetwork, Apache License 2.0).
-func Bicopy(x, y io.ReadWriter, quit <-chan struct{}) {
-	type closeReader interface {
-		CloseRead() error
-	}
-	type closeWriter interface {
-		CloseWrite() error
-	}
-	var wg sync.WaitGroup
-	broker := func(to, from io.ReadWriter) {
-		if _, err := io.Copy(to, from); err != nil {
-			logrus.WithError(err).Debug("failed to call io.Copy")
-		}
-		if fromCR, ok := from.(closeReader); ok {
-			if err := fromCR.CloseRead(); err != nil {
-				logrus.WithError(err).Debug("failed to call CloseRead")
-			}
-		}
-		if toCW, ok := to.(closeWriter); ok {
-			if err := toCW.CloseWrite(); err != nil {
-				logrus.WithError(err).Debug("failed to call CloseWrite")
-			}
-		}
-		wg.Done()
+func broker(dst, src io.ReadWriter, dstName, srcName string) {
+	if _, err := io.Copy(dst, src); err != nil {
+		logrus.WithError(err).Errorf("io.Copy(%s, %s)", dstName, srcName)
 	}
+}
```

Comment:
We don't ensure this today, so it should not block your change. Current code assumes that if we fail to read or write from socket and io.Copy() returns, the other io.Copy() will also returns since it is trying to read of write using the same socket. This may be enough, but I'm not sure.